WorkAndFun/src/main/java/pl/infoshare/FileActions.java

```diff
@@ -1,72 +1,58 @@
         try {
-            particularLines = Files.readString(path).split("\n");
+            FileWriter fw = new FileWriter(path.toString());
+            GSON.toJson(announcements, fw);
+            fw.flush();
+            fw.close();
         } catch (IOException e) {
             e.printStackTrace();
         }
-
-        //fill array line by line, and separate each line into another list, by "," sign
-        for (String particularLine : particularLines) {
-            String[] newString = particularLine.split("~");
-            arrayFromFile.add(particularLine.split("~"));
-        }
-        return arrayFromFile;
     }
 
-    public static void writeToFile(Path path, boolean append, String... parts) {
+    public static void addAnnouncementsToFile(Announcement announcement, Path path) {
+        ArrayList<Announcement> announcements = loadAnnouncementsFromFile(path);
+        announcements.add(announcement);
         try {
-            FileWriter fstream = new FileWriter(String.valueOf(path), append);
-            BufferedWriter out = new BufferedWriter(fstream);
-            if (append) {
-                out.newLine();
-            }
-            String lineToFill = String.join("~", parts);
-
-            out.write(lineToFill);
-            //close buffer writer
-            out.close();
+            FileWriter fw = new FileWriter(path.toString());
+            GSON.toJson(announcements, fw);
+            fw.flush();
+            fw.close();
         } catch (IOException e) {
             e.printStackTrace();
-            System.out.println("Nie znaleziono pliku do zapisu pod wskazaną ścieżką!");
         }
     }
 
-    public static void writeToFileObjectList(List<Announcement> announcementList) {
-        ArrayList<String[]> arrayFromObjectList = new ArrayList<>();
-        writeToFile(Main.ANNOUNCEMENTS_FILE_PATH, false, Announcement.ANNOUNCEMENT_HEADER);
-
-        for (Announcement announcement : announcementList) {
-            arrayFromObjectList.add(announcement.mapToStringArray());
-            writeToFile(Main.ANNOUNCEMENTS_FILE_PATH, true, String.valueOf(announcement.getId()),
-                    String.valueOf(announcement.getType()), String.valueOf(announcement.getServiceType()),
-                    String.valueOf(announcement.getVoivodeship()), String.valueOf(announcement.getCity()),
-                    announcement.getCityDistrict(), announcement.getUnit(), announcement.getNameOfAdvertiser(),
-                    announcement.getPhoneNumber(), announcement.getEmail(), announcement.getDescription(),
-                    announcement.getPrice(), String.valueOf(announcement.getIsPriceNegotiable()),
-                    announcement.getPriceAdditionComment(), String.valueOf(announcement.getDate()),
-                    String.valueOf(announcement.getClientId()), String.valueOf(announcement.getHeader()));
+    public static ArrayList<Announcement> loadAnnouncementsFromFile(Path path) {
```

Comment:
ja bym tylko zmienil nazwe tej metody na readAnnouncementsFromFile, bo skoro wyzej mamy metode writeAnnoun... to analogicznie tutaj powinno byc read.